fix(theming): Adjust dark high contrast to fulfill WCAG 2.1 AAA contrast

susnux opened the pull request:
* <del>:warning: this is chained on #42287 and must be merged first</del>

## Summary
This adds high contrast support for the theming util and unit tests and adjust the high contrast dark theme to fulfill WCAG 2.1 AAA color contrast for text.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

susnux committed:
fix(theming): Adjust dark high contrast to fulfill WCAG 2.1 AAA contrast

Signed-off-by: Ferdinand Thiessen <opensource@fthiessen.de>

susnux commented:
/backport to stable28

JuliaKirschenheuter reviewed (approved):
looks great, thanks!

JuliaKirschenheuter reviewed (changes requested):
Oh no, wait ;(

Dark theme with high contrast on `--color-error`:  the contrast ratio between the focused and unfocused states should be at least 3:1

![Peek 2023-12-19 09-54](https://github.com/nextcloud/server/assets/6078378/5066ec19-18f3-4465-bb1e-ae9ef4aaf49c)

![Screenshot from 2023-12-19 09-53-56](https://github.com/nextcloud/server/assets/6078378/0bf57d00-253f-4ec9-bece-0084361edb14)


We pass it for dark theme:
![Screenshot from 2023-12-19 09-57-24](https://github.com/nextcloud/server/assets/6078378/d4d97039-3262-45fe-b800-ac2762490a84)

susnux commented:
@JuliaKirschenheuter no we do not, the border color is `color-main-text` which is `d8d8d8` on dark theme -> constrast is not 3:1.
We should simply have thicken the border on focus visible or add a second or something :/

susnux committed:
fix(theming): Adjust `color-error` and `color-success` to work for input border colors

Signed-off-by: Ferdinand Thiessen <opensource@fthiessen.de>

susnux commented:
@JuliaKirschenheuter I adjusted all themes for the `color-success` and `color-error` and added a test case for the border contrast. Should be good now.

JuliaKirschenheuter commented:
> @JuliaKirschenheuter I adjusted all themes for the color-success and color-error and added a test case for the border contrast. Should be good now.

That works now, thanks a lot!

susnux commented:
/backport to stable28
